libs/gretty/src/main/groovy/org/akhikhl/gretty/ServerConfigWithInputs.java

```diff
@@ -0,0 +1,172 @@
+package org.akhikhl.gretty;
+
+import groovy.lang.Closure;
+import org.gradle.api.model.ReplacedBy;
+import org.gradle.api.tasks.Console;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputFile;
+import org.gradle.api.tasks.Internal;
+import org.gradle.api.tasks.Optional;
+import org.gradle.api.tasks.PathSensitive;
+import org.gradle.api.tasks.PathSensitivity;
+
+import java.util.List;
+import java.util.Map;
+
+public interface ServerConfigWithInputs {
+    @Optional
+    @Input
+    List<String> getJvmArgs();
+
+    @Optional
+    @Input
+    Map<String, String> getSystemProperties();
+
+    @Optional
+    @Input
+    String getServletContainer();
+
+    @Optional
+    @Input
+    Boolean getManagedClassReload();
+
+    @Optional
+    @Input
+    String getHost();
+
+    @Optional
+    @Input
+    Boolean getHttpEnabled();
+
+    @Optional
+    @Input
+    Integer getHttpPort();
+
+    @Optional
+    @Input
+    Integer getHttpIdleTimeout();
+
+    @Optional
+    @Input
+    Boolean getHttpsEnabled();
+
+    @Optional
+    @Input
+    Integer getHttpsPort();
+
+    @Optional
+    @Input
+    Integer getHttpsIdleTimeout();
+
+    @Optional
+    @Input
+    String getSslHost();
+
+    @Optional
+    @InputFile
+    @PathSensitive(PathSensitivity.NONE)
```

Comment:
Why do you think is `PathSensitivity.NONE` appropriate? Also for the properties of this interface.

Reply:
It seems like the name of the logback config file is not important, only its contents. WDYT?

Reply:
That's something I still need to wrap my head around. Thanks for the explanation, for me it's a long stretch for `PathSensitivity.NONE` to "track file by content, not by name".  Your judgement seems reasonable!